packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -857,7 +857,7 @@ class _NestedScrollController extends ScrollController {
   }
 
   Iterable<_NestedScrollPosition> get nestedPositions sync* {
-    yield* positions;
+    yield* Iterable.castFrom<ScrollPosition, _NestedScrollPosition>(positions);
```

Comment:
weird, why isn't this an instance method, like `positions.castTo<_NestedScrollPosition>()`?

Reply:
Adding instance methods to common interfaces like `Iterable` turned out to be expensive because it requires updating everybody who `implements Iterable`. So original plans on Dart 2 library evolution were changed to use static helpers instead of instance methods. 

/cc @leafpetersen @lrhn

Reply:
The plan was not changed, we are still working on adding the `cast` method.

The `castFrom` static method is a helper method available to anyone who needs to implement `Iterable.cast` because they `implements Iterable`. Since we are using it to implement `Iterable.cast`, and because it's non-breaking, we landed it first.

Reply:
The instance method version should land in the SDK soon.

Reply:
@mraleph based on the above, let's leave a TODO here saying to move from castFrom to cast, so it ends up cleaner

Reply:
done